src/analysis/inner_product_space/rayleigh.lean

```diff
@@ -219,66 +212,71 @@ end
 
 end complete_space
 
+end is_self_adjoint
+
 section finite_dimensional
 variables [finite_dimensional 𝕜 E] [_i : nontrivial E] {T : E →ₗ[𝕜] E}
 
+namespace linear_map
+
+namespace is_symmetric
+
 include _i
 
-/-- The supremum of the Rayleigh quotient of a self-adjoint operator `T` on a nontrivial
+/-- The supremum of the Rayleigh quotient of a symmetric operator `T` on a nontrivial
 finite-dimensional vector space is an eigenvalue for that operator. -/
-lemma has_eigenvalue_supr_of_finite_dimensional (hT : is_self_adjoint T) :
+lemma has_eigenvalue_supr_of_finite_dimensional (hT : T.is_symmetric) :
   has_eigenvalue T ↑(⨆ x : {x : E // x ≠ 0}, is_R_or_C.re ⟪T x, x⟫ / ∥(x:E)∥ ^ 2) :=
 begin
   haveI := finite_dimensional.proper_is_R_or_C 𝕜 E,
-  let T' : E →L[𝕜] E := T.to_continuous_linear_map,
-  have hT' : is_self_adjoint (T' : E →ₗ[𝕜] E) := hT,
+  let T' := hT.to_self_adjoint,
```

Comment:
This also seems like a regression to me?  This file currently does not "really" depend on the adjoint; it just has a spurious dependence introduced by #15281.  But at this line you are proving a fact about a symmetric operator (in finite dimension) by invoking that it's self-adjoint (in the new, adjoint-dependent sense), then at line 245 calling a result `is_self_adjoint.has_eigenvector_of_is_max_on` which is proved for self-adjoint operators by at some point invoking `hT.is_symmetric`, the fact that they are symmetric.

After the renaming PR goes through I hope immediately to separate out `is_symmetric`, and make the Rayleigh file import only that.  If that's to be possible, then this file needs to stay more or less as it was.

Reply:
I guess you could argue that the finite dimension part should use `is_self_adjoint (E →ₗ[𝕜] E)` with finite dimensional adjoint, but it felt to me that that was really splitting hairs.

Reply:
In this case here I was just using the totally overblown Hellinger-Toeplitz theorem to upgrade a linear map to a continuous linear map, so that on can use the things that are defined for `is_self_adjoint (E →L[𝕜] E)`

Reply:
> I guess you could argue that the finite dimension part should use `is_self_adjoint (E →ₗ[𝕜] E)` with finite dimensional adjoint, but it felt to me that that was really splitting hairs.

I was actually arguing the opposite, that the infinite dimension part should use `is_symmetric` ...

Reply:
I know, but the whole point why I made this refactor is that I find that this goes too much against the mathematical standards. Technically you could say that the spectral theorem holds for bounded symmetric operators, but nobody does that. Moreover, the Rayleigh quotient is very useful for unbounded operators, so I would argue that the correct definition would be for `is_self_adjoint (linear_pmap R E F)`, but unfortunately the adjoint is not yet defined for `linear_pmap`